### PR TITLE
Dark mode: Product detail

### LINKED ISSUE #1884: Dark mode: Product detail
Implement light/dark mode on the product detail screen.

### PR DESCRIPTION
**Pull request by 0nko**
Fixes #1884. I've added fading to the image gallery on scroll because I didn't like how it just disappeared before.

|Before|After (light)|After (dark)|
|-|-|-|
|image|image|image|

### CODE REVIEW

File: WooCommerce/src/main/res/drawable/ripple_grey_framed.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
     <item>
         <shape android:shape="rectangle">
-            <solid android:color="@android:color/white" />
+            <solid android:color="@color/card_surface_color" />
```
Comment (AmandaRiu) [inline WooCommerce/src/main/res/drawable/ripple_grey_framed.xml]: This one is a bit tricky since I haven't doing anything with ripple styling yet. There are some missing resource definitions that can be added:
- 1dp dimension - we could properly add it as `minor_10`
- `rippleColor` - we can create a new color in `colors_base.xml` named `color_ripple_overlay` and set it to
  - dark: `#1AFFFFFF` 
  - light: `#1A000000` 

Then change: 
```
<?xml version="1.0" encoding="utf-8"?>
<ripple xmlns:android="http://schemas.android.com/apk/res/android"
    android:color="@color/color_ripple_overlay"
    android:exitFadeDuration="@android:integer/config_shortAnimTime">

    <item>
        <shape android:shape="rectangle">
            <stroke
                android:width="@dimen/minor_10"
                android:color="@color/color_on_surface_medium" />
            <corners android:radius="@dimen/minor_25" />
        </shape>
    </item>

    <item android:id="@android:id/mask">
        <shape android:shape="rectangle">
            <solid android:color="@color/color_surface" />
        </shape>
    </item>
</ripple>
```

File: WooCommerce/src/main/res/layout/fragment_product_detail.xml [hunk 2]
```diff
@@ -21,15 +21,15 @@
             android:layout_width="match_parent"
             android:layout_height="match_parent"
             android:fitsSystemWindows="false"
-            app:contentScrim="?attr/colorPrimary"
-            app:expandedTitleMarginEnd="64dp"
-            app:expandedTitleMarginStart="48dp"
+            android:background="@color/card_surface_color"
```
Comment (AmandaRiu) [inline WooCommerce/src/main/res/layout/fragment_product_detail.xml]: No need for a special color definition - we want to keep it simple. Just set it to `color_surface`

File: WooCommerce/src/main/res/layout/fragment_product_detail.xml [hunk 3]
```diff
@@ -21,15 +21,15 @@
...
+            app:contentScrim="@color/card_surface_color"
             app:layout_scrollFlags="scroll|exitUntilCollapsed">
 
-            <FrameLayout
+            <com.google.android.material.card.MaterialCardView
```
Comment (AmandaRiu) [inline WooCommerce/src/main/res/layout/fragment_product_detail.xml]: ❓ Should this be `MaterialCardView`? It's not really a card.

File: WooCommerce/src/main/res/layout/fragment_product_detail.xml [hunk 4]
```diff
@@ -39,7 +39,6 @@
                     android:layout_gravity="center_horizontal"
                     android:layout_marginTop="@dimen/margin_extra_large"
                     android:layout_marginBottom="@dimen/margin_extra_large"
```
Comment (AmandaRiu) [inline WooCommerce/src/main/res/layout/fragment_product_detail.xml]: This layout file has a bunch of dimens and colors that are still pointing to the old version and not the new design. Also there are also some `TextView`'s that need to be converted to `MaterialTextView` and have the new styles applied. Also `product_detail_status_badge` needs to be updated with new colors and dimens.

File: WooCommerce/src/main/res/layout/product_property_link_view.xml [hunk 5]
```diff
@@ -7,38 +7,38 @@
...
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintTop_toTopOf="parent"
-        app:srcCompat="@drawable/ic_external"/>
+        app:srcCompat="@drawable/ic_external" />
 
     <View
         android:id="@+id/divider"
         style="@style/Woo.Divider"
-        android:layout_marginTop="@dimen/product_detail_property_margin"
+        android:layout_marginTop="@dimen/major_100"
```
Comment (AmandaRiu) [inline WooCommerce/src/main/res/layout/product_property_link_view.xml]: This should have a `android:layout_marginStart="@dimen/major_100` to match how dividers show up in the rest of the designs. This is true for all the dividers on the detail view.

File: WooCommerce/src/main/res/layout/product_property_read_more_view.xml [hunk 6]
```diff
@@ -7,61 +7,59 @@
...
-        android:textColor="@color/wc_purple"
-        android:textSize="@dimen/text_medium"
+        android:textColor="@color/color_secondary"
         android:visibility="gone"
         app:layout_constraintBottom_toTopOf="@id/divider"
         app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintTop_toBottomOf="@+id/textContent"
-        app:layout_goneMarginBottom="@dimen/product_detail_property_margin"
-        tools:visibility="visible"/>
+        app:layout_goneMarginBottom="@dimen/major_100"
```
Comment (AmandaRiu) [inline WooCommerce/src/main/res/layout/product_property_read_more_view.xml]: TIL'd!

File: WooCommerce/src/main/res/layout/product_property_read_more_view.xml [hunk 7]
```diff
@@ -7,61 +7,59 @@
     android:layout_height="wrap_content"
     android:background="?attr/selectableItemBackground">
 
-    <TextView
+    <com.google.android.material.textview.MaterialTextView
```
Comment (AmandaRiu) [inline WooCommerce/src/main/res/layout/product_property_read_more_view.xml]: this layout resource should also use `<merge/>` to remove that extra layer of hierarchy

File: WooCommerce/src/main/res/layout/product_property_read_more_view.xml [hunk 8]
```diff
@@ -7,61 +7,59 @@
...
+        style="@style/Woo.Card.Body"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:layout_marginBottom="@dimen/product_detail_property_margin"
-        android:layout_marginEnd="@dimen/product_detail_property_margin"
-        android:layout_marginStart="@dimen/product_detail_property_margin"
+        android:layout_marginStart="@dimen/major_100"
+        android:layout_marginEnd="@dimen/major_100"
+        android:layout_marginBottom="@dimen/major_100"
         android:background="?attr/selectableItemBackground"
```
Comment (AmandaRiu) [inline WooCommerce/src/main/res/layout/product_property_read_more_view.xml]: Not related to this PR, but this entire view has the ripple effect so this can be removed.

File: WooCommerce/src/main/res/layout/product_property_read_more_view.xml [hunk 9]
```diff
@@ -7,61 +7,59 @@
...
 
     <View
         android:id="@+id/divider"
         style="@style/Woo.Divider"
-        android:layout_marginTop="@dimen/product_detail_property_margin"
+        android:layout_marginTop="@dimen/major_100"
         app:layout_constraintBottom_toBottomOf="parent"
         app:layout_constraintEnd_toEndOf="parent"
-        app:layout_constraintStart_toStartOf="parent"/>
+        app:layout_constraintStart_toStartOf="parent" />
```
Comment (AmandaRiu) [inline WooCommerce/src/main/res/layout/product_property_read_more_view.xml]: This should constrained to the view above it.

File: WooCommerce/src/main/res/layout/product_property_view_horz.xml [hunk 10]
```diff
@@ -7,60 +7,50 @@
     android:layout_height="wrap_content"
     android:background="?attr/selectableItemBackground">
```
Comment (AmandaRiu) [inline WooCommerce/src/main/res/layout/product_property_view_horz.xml]: this layout resource should also use <merge/> to remove that extra layer of hierarchy

File: WooCommerce/src/main/res/layout/product_property_view_horz.xml [hunk 11]
```diff
@@ -7,60 +7,50 @@
...
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        tools:rating="4"
+        tools:visibility="visible" />
 
     <View
         android:id="@+id/divider"
         style="@style/Woo.Divider"
-        android:layout_marginTop="@dimen/product_detail_property_margin"
+        android:layout_marginTop="@dimen/major_100"
```
Comment (AmandaRiu) [inline WooCommerce/src/main/res/layout/product_property_view_horz.xml]: np: this should be constrained to the view above it, and the above view should not be set to align to the parent bottom

**Comment by AmandaRiu** [comment]
@0nko I love the cool table with screenshots. That's a really cool idea! 👍

**Comment by Garance91540** [comment]
> if the entire product detail view is supposed to be a card (ie- have elevation in dark mode) or if it should be flat since there don't really appear to be cards in this view. @Garance91540 WDYT?

The product details screen should be made of 2 cards. The first one with image + description and the other one with additional details (price, inventory, etc)...

File: WooCommerce/src/main/res/layout/product_property_view_horz.xml (hunk 10 above)
Comment (0nko) [reply WooCommerce/src/main/res/layout/product_property_view_horz.xml]: This was necessary for the tap feedback. When I removed it then there was no ripple background applied on Read more click. So I've replaced the `TextView` with a `Button`, since that's what it really was.

File: WooCommerce/src/main/res/layout/product_property_read_more_view.xml (hunk 6 above)
Comment (0nko) [reply WooCommerce/src/main/res/layout/product_property_read_more_view.xml]: Me too :D. It was already there.

File: WooCommerce/src/main/res/layout/product_property_read_more_view.xml (hunk 8 above)
Comment (0nko) [reply WooCommerce/src/main/res/layout/product_property_read_more_view.xml]: This was necessary for the tap feedback. When I removed it then there was no ripple background applied on Read more click. So I've replaced the TextView with a Button, since that's what it really was.

**Comment by jkmassel** [comment]
I'm freezing `3.7`, so this is being bumped to `3.8`. If you need it to be part of the `3.7` release, please re-target `release/3.7` and merge there. Once you have, please ping me that it's been merged and I'll be happy to cut a new release candidate!

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/WCProductPropertyCardView.kt [hunk 12]
```diff
@@ -13,7 +14,7 @@ class WCProductPropertyCardView @JvmOverloads constructor(
     context: Context,
     attrs: AttributeSet? = null,
     defStyle: Int = 0
-) : androidx.cardview.widget.CardView(context, attrs, defStyle) {
+) : FrameLayout(context, attrs, defStyle) {
```
[the remark this answers is not in the document]
Comment (0nko) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/WCProductPropertyCardView.kt]: No, this is actually the card that holds the individual property rows.

File: WooCommerce/src/main/res/drawable/ic_arrow_right.xml [hunk 13]
```diff
@@ -0,0 +1,10 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
[the remark this answers is not in the document]
Comment (0nko) [reply WooCommerce/src/main/res/drawable/ic_arrow_right.xml]: > Also, I believe this should actually use @color/color_secondary, but @Garance91540 can verify.

I don't think so. It's the little chevron used on the right edge of elements.

File: WooCommerce/src/main/res/layout/product_property_view_vert.xml [hunk 14]
```diff
@@ -12,56 +11,53 @@
...
         android:id="@+id/ratingBar"
-        style="@style/Base.Widget.AppCompat.RatingBar.Small"
+        style="@style/Woo.RatingsBar.Small.Colored"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
         android:isIndicator="true"
         android:numStars="5"
-        android:progressTint="@color/grey_darken_30"
         android:rating="0"
         android:stepSize="1"
```
Comment (AmandaRiu) [inline WooCommerce/src/main/res/layout/product_property_view_vert.xml]: np: All three properties are covered by the style so they can be removed. This ensures that if we update the style later the changes will properly be reflected across the app. Same is true for the other file: `product_property_view_horz.xml`.

File: WooCommerce/src/main/res/layout/product_property_editable_view.xml [hunk 15]
```diff
@@ -4,14 +4,15 @@
     android:layout_width="match_parent"
     android:layout_height="wrap_content">
 
-    <EditText
+    <androidx.appcompat.widget.AppCompatEditText
```
[the remark this answers is not in the document]
Comment (0nko) [reply WooCommerce/src/main/res/layout/product_property_editable_view.xml]: The `TextInputLayout` will need to be updated before dark mode is released because the label is a bit messed up right now:

image

File: WooCommerce/src/main/res/layout/fragment_product_detail.xml (hunk 3 above)
Comment (0nko) [reply WooCommerce/src/main/res/layout/fragment_product_detail.xml]: You're right. Initially I couldn't get the image gallery background to display properly without it but eventually I found the solution.

File: WooCommerce/src/main/res/layout/product_property_cardview.xml [hunk 16]
```diff
@@ -1,26 +1,27 @@
 <?xml version="1.0" encoding="utf-8"?>
-<androidx.cardview.widget.CardView
+<com.google.android.material.card.MaterialCardView
```
[the remark this answers is not in the document]
Comment (0nko) [reply WooCommerce/src/main/res/layout/product_property_cardview.xml]: This was tricky because I couldn't get the correct "material" style working without adding the extra layer of view groups. I believe the reason was thast we're creating the `WCProductPropertyCardView` in code and adding it to the view hierarchy manually and the styles weren't being applied properly. I've solved it by inflating the custom view in the fragment instead instead of instantiating it directly.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductDetailFragment.kt [hunk 17]
```diff
@@ -93,6 +97,13 @@ class ProductDetailFragment : BaseProductFragment(), OnGalleryImageClickListener
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         initializeViewModel()
+
+        val content = view.findViewById<View>(R.id.collapsing_content)
+        val appBar = view.findViewById<AppBarLayout>(R.id.app_bar_layout)
+        appBar.addOnOffsetChangedListener(OnOffsetChangedListener { appBarLayout, verticalOffset ->
+            val scrollRange = (appBarLayout.totalScrollRange - DisplayUtils.dpToPx(context, 64)).toFloat()
+            content.alpha = 1.0f - abs(verticalOffset / scrollRange)
+        })
```
[the remark this answers is not in the document]
Comment (0nko) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductDetailFragment.kt]: I just realized there already was an animation but I had disabled all of them on the emulator for connected tests 🤦‍♂

File: WooCommerce/src/main/res/layout/fragment_product_detail.xml (hunk 2 above)
Comment (0nko) [reply WooCommerce/src/main/res/layout/fragment_product_detail.xml]: This doesn't work in dark mode:

image

The image gallery is not a material card but we still want to have the same background.

File: WooCommerce/src/main/res/layout/product_property_view_vert.xml (hunk 14 above)
Comment (0nko) [reply WooCommerce/src/main/res/layout/product_property_view_vert.xml]: > That shouldn't be necessary. As long as we just have two cards like Garance mentioned, then we really should be able to create two distinct UI MaterialCardViews in the layout and then just add the various properties to the appropriate card and not worry about inserting manual dividers. 

This doesn't work automatically. Even though the `Woo.Card` style is applied to the `WCProductPropertyCardView`, the dividers are missing 🤔

**Comment by 0nko** [comment]
> The product details screen should be made of 2 cards. The first one with image + description and the other one with additional details (price, inventory, etc)...

@Garance91540 What about the title, where should that go? Currently, the purchase details is its own card with a caption (see the screenshot). Should the title be removed?

image

Also, what should the color of the right arrows be for Inventory and Shipping?

**Comment by Garance91540** [comment]
Heya @0nko !

In the new designs, there shouldn't be a need for the "purchase details" card anymore. All content in this card should be reintroduced as part of the new sections of the product card (ie: "shipping" now has it's own "Shipping" list item element. 

I see in your screenshots you have a "Purchase details" > "Purchase notes" card, but I'm not quite sure  what it's referring to? Shouldn't this be on the Order screen instead of the product card?

>Also, what should the color of the right arrows be for Inventory and Shipping?

In the final designs, there shouldn't be arrows on the cards. I brought it up in a previous PR (can't find it) and @nbradbury  or @anitaa1990 told me they should be temporary while we were implementing them (is that still the case?).

Screen Shot 2020-02-26 at 12 28 32 PM

Side note @0nko , on the screenshot above, the icon for the shipping list item is mostly outline vs the one you shared with me seems filled up. Is it possible to update it?

File: WooCommerce/src/main/res/layout/product_property_editable_view.xml (hunk 15 above)
[the remark this answers is not in the document]
Comment (0nko) [reply WooCommerce/src/main/res/layout/product_property_editable_view.xml]: I've updated the `TextInputLayout` style to fix some padding and color issues but I'd like to get it approved by @Garance91540.

File: WooCommerce/src/main/res/layout/fragment_product_detail.xml (hunk 2 above)
Comment (AmandaRiu) [reply WooCommerce/src/main/res/layout/fragment_product_detail.xml]: Ah I see what you mean. I'm really worried about having a color asset called `card_surface_color` because then it may get used as background for cards - but that's all handled by the OS. Looking at the Elevation card in the "System - Material Dark Components" figma designs elevation of 1dp is defined at .5 alpha and if you click on that component it will give you the actual color to use: `0DFFFFFF`:

Screen Shot 2020-02-26 at 12 15 48 PM

 I plugged that in and it worked great. To avoid confusion, how about we add a new color to the `wc_colors_base.xml` file (that's where raw theme colors are defined):

```
<color name="woo_white_alpha_005">#0DFFFFFF</color>
```

Then add a new color in `colors_base` for light/dark versions:
```
<!-- dark version -->
<color name="color_elevation_overlay_01">@color/woo_white_alpha_005</color>

<!-- light version -->
<color name="color_elevation_overlay_01">@color/color_surface</color>
```

This way the definitions are generic and reusable - and we can expand on it easily by adding other elevation values as needed - for example: `color_elevation_overlay_02`, `color_elevation_overlay_03`... WDYT?

File: WooCommerce/src/main/res/layout/product_property_cardview.xml [hunk 18]
```diff
@@ -10,6 +10,7 @@
         android:id="@+id/cardContainerView"
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
+        android:layout_marginBottom="-1dp"
```
Comment (AmandaRiu) [inline WooCommerce/src/main/res/layout/product_property_cardview.xml]: Why is this needed?

File: WooCommerce/src/main/res/layout/product_property_link_view_layout.xml [hunk 19]
```diff
@@ -0,0 +1,43 @@
...
+    <ImageView
+        android:id="@+id/iconExternal"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_margin="@dimen/major_100"
+        android:importantForAccessibility="no"
+        android:tint="@color/color_secondary"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        app:srcCompat="@drawable/ic_external" />
```
Comment (AmandaRiu) [inline WooCommerce/src/main/res/layout/product_property_link_view_layout.xml]: np: Both of these should have their bottoms constrained to the divider with `app:layout_constraintBottom_toTopOf="@+id/divider"`

File: WooCommerce/src/main/res/values/styles.xml [hunk 20]
```diff
@@ -432,25 +432,19 @@ Use this file to override styles in the styles_base.xml file.
     </style>
 
     <style name="Woo.MaterialComponents.EditText" parent="@style/Widget.MaterialComponents.TextInputLayout.OutlinedBox">
-        <item name="android:textColor">@color/wc_black_darker</item>
-        <item name="android:textColorHint">@color/wc_black_lighter</item>
-        <item name="android:textSize">@dimen/text_large</item>
+        <item name="android:textColor">@color/color_on_surface_high</item>
+        <item name="android:textColorHint">@color/color_on_surface_medium</item>
+        <item name="android:textSize">@dimen/text_minor_125</item>
         <item name="colorControlNormal">@color/list_divider</item>
```
Comment (AmandaRiu) [inline WooCommerce/src/main/res/values/styles.xml]: I'm not sure what component is using this style, but these are the old styles and we shouldn't be editing them with new logic.  There are existing TextInputLayout styles defined in `styles_base.xml`.

File: WooCommerce/src/main/res/layout/product_property_read_more_view_layout.xml [hunk 21]
```diff
@@ -0,0 +1,61 @@
...
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/textCaption"
+        tools:text="textContent" />
+
+    <com.google.android.material.button.MaterialButton
+        android:id="@+id/btnReadMore"
+        style="@style/Woo.Button"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="@dimen/minor_100"
```
Comment (AmandaRiu) [inline WooCommerce/src/main/res/layout/product_property_read_more_view_layout.xml]: np: The text on this button is not aligned with the text in the card. Recommend setting `layout_marginStart` to 0dp. 

Screen Shot 2020-02-26 at 12 34 45 PM

File: WooCommerce/src/main/res/layout/product_property_view_horz_layout.xml [hunk 22]
```diff
@@ -0,0 +1,58 @@
...
+        android:id="@+id/textPropertyValue"
+        android:layout_width="wrap_content"
+        style="@style/Woo.TextView.Subtitle1"
+        android:layout_height="wrap_content"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toStartOf="@id/ratingBar"
+        app:layout_constraintTop_toTopOf="parent"
+        android:ellipsize="end"
+        android:maxLines="1"
+        tools:text="textDetail"/>
```
Comment (AmandaRiu) [inline WooCommerce/src/main/res/layout/product_property_view_horz_layout.xml]: These two views are not constrained against each other which leads to unintentional overlap of the view if the text is longer than expected:

Screen Shot 2020-02-26 at 1 32 51 PM

To fix this you can add this:
```
app:layout_constraintEnd_toStartOf="@+id/textPropertyValue" 
```
and then set the width to 0dp so it will fill the space. So that first text view would look like this if we want to force it to only a single line:
```
<com.google.android.material.textview.MaterialTextView
        android:id="@+id/textPropertyName"
        style="@style/Woo.TextView.Subtitle1"
        android:layout_width="0dp"
        android:layout_height="wrap_content"
        android:layout_margin="@dimen/major_100"
        app:layout_constraintBottom_toBottomOf="parent"
        app:layout_constraintEnd_toStartOf="@+id/textPropertyValue"
        app:layout_constraintStart_toStartOf="parent"
        app:layout_constraintTop_toTopOf="parent"
        android:ellipsize="end"
        android:singleLine="true"
        tools:text="textCaption tis is really long text that will wrap." />
``` 
which will give us:
Screen Shot 2020-02-26 at 1 39 04 PM

or we can allow text to wrap with this:
```
<com.google.android.material.textview.MaterialTextView
        android:id="@+id/textPropertyName"
        style="@style/Woo.TextView.Subtitle1"
        android:layout_width="0dp"
        android:layout_height="wrap_content"
        android:layout_margin="@dimen/major_100"
        app:layout_constraintBottom_toBottomOf="parent"
        app:layout_constraintEnd_toStartOf="@+id/textPropertyValue"
        app:layout_constraintStart_toStartOf="parent"
        app:layout_constraintTop_toTopOf="parent"
        tools:text="textCaption tis is really long text that will wrap." />
```
which gives us: 
Screen Shot 2020-02-26 at 1 40 03 PM

```

Also all of the views above the divider should have their bottoms constrained to the top of the divider with `layout_constraintBottom_toTopOf="@+id/divider"`

File: WooCommerce/src/main/res/layout/product_property_view_vert_layout.xml [hunk 23]
```diff
@@ -1,67 +1,62 @@
...
         android:id="@+id/textPropertyName"
+        style="@style/Woo.TextView.Subtitle1"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:layout_marginStart="@dimen/product_detail_property_margin"
-        android:layout_marginTop="@dimen/product_detail_property_margin"
-        android:textColor="@color/wc_black_darker"
-        android:textSize="@dimen/text_large"
+        android:layout_marginStart="@dimen/major_100"
+        android:layout_marginTop="@dimen/major_100"
```
Comment (AmandaRiu) [inline WooCommerce/src/main/res/layout/product_property_view_vert_layout.xml]: Just like the horizontal variation, you'll want to make sure your TextViews are constrained properly so they layout properly on different screen sizes and various lengths of text. Also, note that the divider is at the bottom of the view so other views directly above it should have their bottoms constrained to the top of the divider.

File: WooCommerce/src/main/res/layout/product_property_read_more_view.xml [hunk 24]
```diff
@@ -57,6 +57,6 @@
         android:layout_marginStart="@dimen/major_100"
         app:layout_constraintBottom_toBottomOf="parent"
         app:layout_constraintEnd_toEndOf="parent"
-        app:layout_constraintStart_toStartOf="parent" />
+        app:layout_constraintStart_toStartOf="@id/btnReadMore" />
```
[the remark this answers is not in the document]
Comment (0nko) [reply WooCommerce/src/main/res/layout/product_property_read_more_view.xml]: It was: https://github.com/woocommerce/woocommerce-android/pull/1962#discussion_r381810927 :)

File: WooCommerce/src/main/res/layout/product_property_editable_view_layout.xml [hunk 25]
```diff
@@ -0,0 +1,31 @@
...
+    <View
+        android:id="@+id/divider"
+        style="@style/Woo.Divider"
+        android:layout_marginTop="@dimen/major_100"
+        android:layout_marginStart="@dimen/major_100"
+        app:layout_constraintTop_toBottomOf="@id/editText"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="@id/edit_text_input" />
+
+</merge>
```
[the remark this answers is not in the document]
Comment (0nko) [reply WooCommerce/src/main/res/layout/product_property_editable_view_layout.xml]: This must have happened when I merged the changes from `feature/darkmode` because it definitely didn't look like this :). I should have checked.. Thanks for catching it.

File: WooCommerce/src/main/res/layout/product_property_cardview.xml (hunk 18 above)
Comment (0nko) [reply WooCommerce/src/main/res/layout/product_property_cardview.xml]: Every property view has a divider at the bottom. I've tried to make it work by hiding the last one but because views can be added & removed it added a lot of complicated logic. This is a lot simpler. I'll add a comment about it.

File: WooCommerce/src/main/res/layout/product_property_read_more_view_layout.xml (hunk 21 above)
Comment (0nko) [reply WooCommerce/src/main/res/layout/product_property_read_more_view_layout.xml]: Also another merge problem it seems.

File: WooCommerce/src/main/res/layout/product_property_editable_view_layout.xml (hunk 25 above)
[the remark this answers is not in the document]
Comment (0nko) [reply WooCommerce/src/main/res/layout/product_property_editable_view_layout.xml]: Actually it's because I removed the `TextInputLayout` (which was anchored), when I tried to make the edit text look like the design above.

**Comment by 0nko** [comment]
Thanks @AmandaRiu 🙇. Ready for another round.

**Comment by 0nko** [comment]
@Garance91540 I've talked to @AmandaRiu about the screen changes (2 cards with updated properties) and we've agreed to keep the PR limited to style changes. We'll update the cards in a different PR when we start working on product editing.

File: WooCommerce/src/main/res/layout/fragment_product_detail.xml (hunk 2 above)
Comment (0nko) [reply WooCommerce/src/main/res/layout/fragment_product_detail.xml]: Almost missed this comment. This makes sense, thanks! 👍

File: WooCommerce/src/main/res/layout/product_property_read_more_view.xml (hunk 24 above)
[the remark this answers is not in the document]
Comment (AmandaRiu) [reply WooCommerce/src/main/res/layout/product_property_read_more_view.xml]: Ah I think there was a misunderstanding there. When I said "anchored to the view on top of it" I meant that the bottom text view was anchored to the bottom of the container and so was the divider, but with constraintlayout you want to have the text view anchored to the top of the divider not the bottom of the container - so `layout_constraintTop_toBottomOf` is what I was looking to get changed not the start of the anchor.

[the remark this answers is not in the document]
Comment (AmandaRiu) [reply WooCommerce/src/main/res/layout/product_property_read_more_view.xml]: To summarize, I was talking about the `Top` and `Bottom` anchors, not the `Start` or `End` anchors, which is what this specific code comment was referring to.

File: WooCommerce/src/main/res/layout/product_property_cardview.xml (hunk 18 above)
Comment (AmandaRiu) [reply WooCommerce/src/main/res/layout/product_property_cardview.xml]: Ooh that's interesting. Cool.
